lib/Connection.php

```diff
@@ -249,7 +256,7 @@ private function closeSocket() {
         $this->outputBuffer = "";
         $this->outputBufferLength = 0;
 
-        if (is_resource($this->socket)) {
+        if (!$this->persistent && is_resource($this->socket)) {
```

Comment:
If I do not call the close function, the progress will run forever,  and it won't execute the code after the amp\run() function. So I must call the close. 
If I call the` fclose()`, the persistent connection will lost. 
So...